components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -418,11 +418,15 @@ def pickFirstSignature(condition, filterLambda):
                     template = info.template
                     declType = info.declType
 
+                    argName = "arg%d" % distinguishingIndex
+
                     testCode = instantiateJSToNativeConversionTemplate(
                         template,
                         {"val": distinguishingArg},
                         declType,
-                        "arg%d" % distinguishingIndex)
+                        argName)
+                    if type_needs_auto_root(type):
+                        testCode.append(CGGeneric("auto_root!(in(cx) let %s = %s);" % (argName, argName)))
```

Comment:
What if we move this into `instantiateJSToNativeConversionTemplate` instead? Can we also add codegen tests for:
* two overloads - ArrayBuffer and DOMString
* three overloads - a union containing ArrayBuffer and DOMString

Reply:
Specifically, if we add a field to JSToNativeConversionInfo and make getJSToNativeConversionInfo set it appropriately, instantiateJSToNativeConversionTemplate could add the extra auto_root! declaration.

Reply:
Ah, now that I read your reply, it seems I misread it. Do you think https://github.com/servo/servo/pull/20396/commits/01c6bee2437946a3e06a05da9775ca11218a67bc is sufficient for now or should I change it further?

Reply:
I'm a bit worried that we'll keep hitting similar issues with missing CustomAutoRoot wrappers if we keep doing spot-fixes that only change a single call-site at a time.

Reply:
Yeah, I agree completely.

However in this particular case I'm not sure what might be the best approach.
In general `getJSToNativeConversionInfo` returns template, declType and default, which is then later passed manually to `instantiateJSToNativeConversionTemplate`.

In the updated https://github.com/servo/servo/pull/20396/commits/01c6bee2437946a3e06a05da9775ca11218a67bc, for the `instantiateJSToNativeConversionTemplate` call I'm passing the information if the intantiated template needs the rooting (which is later done inside the call).

If I'm not mistaken, do you mean that we should return an additional field (e.g. `needsAutoRooting`) from `getJSToNativeConversionInfo` and pass it later manually for every following `instantiateJSToNativeConversionTemplate` call if needed, as well?

Reply:
Bleah. We should really refactor the code generation to pass the JSToNativeConversionInfo class directly to instantiateJSToNativeConversionTemplate so it's easier to make changes like this. I'll file a separate issue to do that and clean up the changes from this PR.

Reply:
Filed https://github.com/servo/servo/issues/20403.